lib/ecto/query/builder.ex

```diff
@@ -217,6 +217,11 @@ defmodule Ecto.Query.Builder do
     {expr, params_acc}
   end
 
+  def escape({:=, _, _}, _type, _params_acc, _vars, _env) do
+    error! "match operator is not supported: `=`. " <>
```

Comment:
The function error would add `Macro.to_string` representation of the whole offending part of the query. Should we do something similar here? It might be hard to find the exact point in some complex expression and this way we're able to show the surroundings of where's the issue.